# Computational Essay #1
## Jungyu Baeck, Blake Sodikoff

### Problem 1

Background: tables of stellar brightness list the flux density for stars of designated magnitude. In observational astronomy, one of the most fundamental questions we ask about any celestial object is: how much light does it emit? While brightness is often expressed in magnitudes or flux units, a deeper physical understanding comes from thinking in terms of photons, the individual quanta of light. Since every telescope detector counts photons, knowing how many photons arrive from a source at a given wavelength is crucial for designing observations, estimating exposure times, and interpreting data. This essay presents a computational approach to calculating the photon arrival rate



In observational astronomy, stellar brightness is often listed as **flux density per unit frequency** $F_\nu$ in units as 
$$
{W\,m^{-2}\,Hz^{-1}}
$$
However, many physical calculations and analyses require the ***flux density per unit wavelength** $F_\lambda$ in units as
$$
{erg\,s^{-1}\,cm^{-2}\,\text{Å}^{-1}}
$$


We will need to convert $F_\nu$ into $F_\lambda$. We do it this way:

We start from the equality of flux densities:

$$
F_\nu\, d\nu = F_\lambda\, d\lambda
$$

Using the relation between frequency and wavelength:

$$
\nu = \frac{c}{\lambda}
$$

we get the derivative:

$$
\frac{d\nu}{d\lambda} = -\frac{c}{\lambda^2}
$$

Thus, substituting back into the flux equation:

$$
F_\lambda = F_\nu \frac{d\nu}{d\lambda} = F_\nu \frac{c}{\lambda^2}
$$

We have:
$$
F_\nu = F_\lambda \frac{\lambda^2}{c}
$$

Since we want $F_\lambda$ in cgs units we include conversion factors:
$${W} = 10^{7}\,\mathrm{erg\,s^{-1}}$$

$${m^2} = 10^{4}\,\mathrm{cm^2}$$

$${m} = 10^{10}\,\text{Å}$$



The following function does it

In [14]:
# Define function.

def convert_fnu(fnu,w) :
    """
    Returns a value of F-lambda given F-nu
    and a wavelength.
    
    Arguments (required)
    1. F-nu in units of Watt per square m per Hz
    2. Wavelength in Angstrom
    
    Returns
    
    F-lambda in units of erg per second per square 
    cm per Angstrom
    """
    c = 3.0E+8  # m/s
    
    lam_m = w * 1e-6 # Convert wavelength from microns to meters
    fnu_si = fnu * 1e-26 # Convert Fν from Jy to W m^-2 Hz^-1 (1 Jy = 1e-26 W m^-2 Hz^-1)
    
    result = fnu_si * c / (lam_m ** 2)
    result = result * 1.0E+7 / 1.0E+4 / 1.0E+10 
    # Convert to erg s^-1 cm^-2 Å^-1
    # 1 W = 10^7 erg/s, 1 m² = 10^4 cm², 1 m = 10^10 Å
    return result

In [15]:
help(convert_fnu)

Help on function convert_fnu in module __main__:

convert_fnu(fnu, w)
    Returns a value of F-lambda given F-nu
    and a wavelength.
    
    Arguments (required)
    1. F-nu in units of Watt per square m per Hz
    2. Wavelength in Angstrom
    
    Returns
    
    F-lambda in units of erg per second per square 
    cm per Angstrom



In [16]:
Filters = ['U', 'B', 'V', 'R', 'I', 'J', 'H', 'K', 'g', 'r', 'i', 'z']
Wavelengths = [0.366, 0.436, 0.545, 0.641, 0.798, 1.26, 1.60, 2.20, 0.52, 0.67, 0.79, 0.91]  # in microns
Fluxes = [1810, 4260, 3640, 3080, 2550, 1600, 1080, 670, 3730, 4490, 4760, 4810]  # in Jy

LFluxes = [convert_fnu(f, w) for f, w in zip(Fluxes, Wavelengths)]


for name, f, w, fl in zip(Filters, Fluxes, Wavelengths, LFluxes):
    print(f" Band:{name}   Wavelength:{w:.3f}       F(in Jy):{f:.2e}       F(in erg s⁻¹ cm⁻² Å⁻¹):{fl:.3e}")

 Band:U   Wavelength:0.366       F(in Jy):1.81e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):4.054e-09
 Band:B   Wavelength:0.436       F(in Jy):4.26e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):6.723e-09
 Band:V   Wavelength:0.545       F(in Jy):3.64e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):3.676e-09
 Band:R   Wavelength:0.641       F(in Jy):3.08e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):2.249e-09
 Band:I   Wavelength:0.798       F(in Jy):2.55e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):1.201e-09
 Band:J   Wavelength:1.260       F(in Jy):1.60e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):3.023e-10
 Band:H   Wavelength:1.600       F(in Jy):1.08e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):1.266e-10
 Band:K   Wavelength:2.200       F(in Jy):6.70e+02       F(in erg s⁻¹ cm⁻² Å⁻¹):4.153e-11
 Band:g   Wavelength:0.520       F(in Jy):3.73e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):4.138e-09
 Band:r   Wavelength:0.670       F(in Jy):4.49e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):3.001e-09
 Band:i   Wavelength:0.790       F(in Jy):4.76e+03       F(in erg s⁻¹ cm⁻² Å⁻¹):2.288e-09
 Band:z   

Astronomical detectors count photons, not directly energy. Therefore, knowing the photon arrival rate per unit wavelength is very useful. The energy of a single photon is given by:
$$
E_\gamma = \frac{hc}{\lambda}
$$
The photon rate per unit area per unit wavelength is then:
$$
N_\lambda = \frac{F_\lambda}{E_\gamma} = \frac{F_\lambda \, \lambda}{h c}
$$


$F_\lambda$ = flux density (erg s⁻¹ cm⁻² Å⁻¹)

$\lambda$ = wavelength (Å)

h = Planck’s constant = 6.626×$10^{-27}$ erg·s

c = 2.998x$10^{18}$ Å/s

The following function does it

In [17]:
def photon_rate(Flambda, w):
    """
    Calculate photon rate (photons/s/cm^2/Å)

    Parameters
    ----------
    Flambda : float
        Flux density (erg/s/cm^2/Å)
    w : float
        Wavelength (microns)

    Returns
    -------
    N_lambda : float
        Photon arrival rate per Å
    """
    h = 6.626E-27       # erg·s
    c = 2.998E+18       # Å/s
    lam_A = w * 1.0E+4  # microns -> Å
    return Flambda * lam_A / (h * c)

Then, we can calculate Photon rate in cm⁻² s⁻¹ Å⁻¹

In [18]:
F_lambdas = [convert_fnu(f, w) for f, w in zip(Fluxes, Wavelengths)]
Photon_rates = [photon_rate(fl, w) for fl, w in zip(F_lambdas, Wavelengths)]


for name, w, f, fl, N in zip(Filters, Wavelengths, Fluxes, F_lambdas, Photon_rates):
    print(f"  Band:{name}   Wavelength:{w:.3f}         F(in Jy):{f:.2e}         F(in erg s⁻¹ cm⁻² Å⁻¹):{fl:.3e}        Photon rate(in cm⁻² s⁻¹ Å⁻¹):{N:.3e}")

  Band:U   Wavelength:0.366         F(in Jy):1.81e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):4.054e-09        Photon rate(in cm⁻² s⁻¹ Å⁻¹):7.469e+02
  Band:B   Wavelength:0.436         F(in Jy):4.26e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):6.723e-09        Photon rate(in cm⁻² s⁻¹ Å⁻¹):1.476e+03
  Band:V   Wavelength:0.545         F(in Jy):3.64e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):3.676e-09        Photon rate(in cm⁻² s⁻¹ Å⁻¹):1.009e+03
  Band:R   Wavelength:0.641         F(in Jy):3.08e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):2.249e-09        Photon rate(in cm⁻² s⁻¹ Å⁻¹):7.257e+02
  Band:I   Wavelength:0.798         F(in Jy):2.55e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):1.201e-09        Photon rate(in cm⁻² s⁻¹ Å⁻¹):4.826e+02
  Band:J   Wavelength:1.260         F(in Jy):1.60e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):3.023e-10        Photon rate(in cm⁻² s⁻¹ Å⁻¹):1.918e+02
  Band:H   Wavelength:1.600         F(in Jy):1.08e+03         F(in erg s⁻¹ cm⁻² Å⁻¹):1.266e-10        Photon rate(in cm⁻² s⁻¹ Å⁻¹):1.019e+02
  Band:K   Wa

After performing these calculations for each photometric band, we find that shorter wavelength bands tend to have lower photon rates because photons are more energetic. Longer wavelength bands have higher photon rates per unit flux because each photon carries less energy, so more photons are needed to carry the same flux. Compared with Given V-Band values, $F_\lambda$ = 3.68E-09 and Photon rate=1008, our results are so close with them, tiny rounding error. This problem highlighted the importance of careful unit conversions when working across different physical quantities and measurement systems. Overall, these conversions are fundamental tools in observational astronomy. This enable accurate interpretation of stellar brightness and detector performance.

#### Problem 2 -- How Bright is the Night Sky?

##### Instructions:

    Explore the effect of the moon on observations of faint objects. Quoted below is
    the sky brightness in magnitudes per square arcsecond. That means that over a 1
    arcsecond patch of the sky, the sky delivers photons at the same rate as a star of
    the stated magnitude.

    How much brighter is the sky at quarter moon (phase = 7) compared to new
    moon (phase = 0)? How much brighter is the sky at full moon compared to new
    moon? Do this for each filter in this table.

We want to compare how much brighter the night sky becomes as the Moon's phase changes. The table gives sky brightness in magnitudes per square arcsecond, which means a 1"x1" patch of sky is as bright as a star of that magnitude. 

We will compute, for each filter (U, B, V, R, I): 

    -The brightness factor at quarter moon (phase 7) to new moon (phase 0)
    -The brightness factor at full moon (phase 14) to new moon (phase 0)
    

Brightness in astronomy uses a logarithmic magnitude scale. If $m_1$ and $m_2$ are magnutudes and $F_1$, $F_2$ are the corresponding fluxes (brightness): 

$\frac{F_1}{F_2} \;=\; 10^{0.4\,(m_2 - m_1)}$


So for a given filter: 


$\text{Brightness ratio} \;=\; \frac{F_{\text{phase}}}{F_{\text{new}}}
\;=\; 10^{0.4\,(m_{\text{new}} - m_{\text{phase}})}$


Smaller magnitude $\Rightarrow$ brighter sky (because it's a logarithmic, inverted scale). 

In [1]:
import numpy as np
import pandas as pd

In [2]:
def brightness_ratio(m_new: float, m_phase: float) -> float:
    """
    Return the multiplicative brightness factor of the sky
    at a given phase relative to new moon (phase=0),
    using the astronomical magnitude–flux relation.

    Parameters
    ----------
    m_new : float
        Sky magnitude at new moon (mag/arcsec^2).
    m_phase : float
        Sky magnitude at some other lunar phase (mag/arcsec^2).

    Returns
    -------
    float
        Brightness ratio F_phase / F_new (dimensionless).
    """
    # Magnitude difference formula: F1/F2 = 10^(0.4 * (m2 - m1))
    # Here: F_phase / F_new = 10^(0.4 * (m_new - m_phase))
    return 10 ** (0.4 * (m_new - m_phase))

In [3]:
# Filters in order
filters = ["U", "B", "V", "R", "I"]

In [4]:
# Magnitudes (mag/arcsec^2) at each phase, by filter (order must match `filters`)
mags_new   = [22.0, 22.7, 21.8, 20.9, 19.9]  # phase 0 (new)
mags_quart = [19.9, 21.6, 21.4, 20.6, 19.7]  # phase 7 (quarter)
mags_full  = [17.0, 19.5, 20.0, 19.9, 19.2]  # phase 14 (full)

In [5]:
# Computing brightness ratios with list comprehensions
ratio_quarter_over_new = [brightness_ratio(mn, mq) for mn, mq in zip(mags_new, mags_quart)]
ratio_full_over_new    = [brightness_ratio(mn, mf) for mn, mf in zip(mags_new, mags_full)]

In [6]:
# Computing magnitude differences for intuition
delta_m_quarter = [mn - mq for mn, mq in zip(mags_new, mags_quart)]
delta_m_full    = [mn - mf for mn, mf in zip(mags_new, mags_full)]

In [7]:
# Assembling a readable table
df = pd.DataFrame({
    "Filter": filters,
    "New Moon (mag)": mags_new,
    "Quarter Moon (mag)": mags_quart,
    "Full Moon (mag)": mags_full,
    "Δm (Quarter - New)": delta_m_quarter,   # positive means quarter is brighter
    "Δm (Full - New)": delta_m_full,         # positive means full is brighter
    "Quarter/New Ratio": ratio_quarter_over_new,
    "Full/New Ratio": ratio_full_over_new
})

In [8]:
# Rounds ratios for display; keep delta m to 2 decimals for readability
df_rounded = df.copy()
df_rounded["Δm (Quarter - New)"] = df_rounded["Δm (Quarter - New)"].round(2)
df_rounded["Δm (Full - New)"]    = df_rounded["Δm (Full - New)"].round(2)
df_rounded["Quarter/New Ratio"]  = df_rounded["Quarter/New Ratio"].round(2)
df_rounded["Full/New Ratio"]     = df_rounded["Full/New Ratio"].round(2)

df_rounded

,Filter,New Moon (mag),Quarter Moon (mag),Full Moon (mag),Δm (Quarter - New),Δm (Full - New),Quarter/New Ratio,Full/New Ratio
0,U,22.0,19.9,17.0,2.1,5.0,6.92,100.00
1,B,22.7,21.6,19.5,1.1,3.2,2.75,19.05
2,V,21.8,21.4,20.0,0.4,1.8,1.45,5.25
3,R,20.9,20.6,19.9,0.3,1.0,1.32,2.51
4,I,19.9,19.7,19.2,0.2,0.7,1.20,1.91


## Validation / Intuition

*U Band example:*

$\Delta m = 22.0 - 19.9 = 2.1 \Rightarrow 10^{0.4\times 2.1} \approx 6.92$ (quarter/new). 

For full moon: $\Delta m = 22.0 - 17.0 = 5.0 \Rightarrow 10^{0.4\times 5.0} = 100$. 

Similar checks hold true in each band. 


Key point: The short wavelength bands like U and B brighten the most with Moon light (rayeigh scattering makes the blue/UV sky shine more) while R and I are less affected. That's why faint, blue sensitive observations are best at new moon. 

## Conclusion

The magnitude scale is logarithmic: a small change in magnitude means a large change in brightness. Using the assignment's given sky brightness table, the sky becomes roughly 1.2-6.9 times brighter at quarter moon (depending on the filter) and up to 100 times brighter at full moon in the U band, with smaler but still significant brightening in B and much smaller in R and I. Practically, this means observing very faint targets is most efficient near new moon, especially in the shorter wavelength filters where scattered monlight impacts the sky background the most. 

#### Problem 3 -- Observing Faint Objects

##### Overview

You have a star which has $F_\nu = 0.0364 \,\text{mJy}$ at all wavelengths. 


Goals: 

    -What is the magnitude of this star in the UBVRI filters? 
    -How bright is the star compared to the sky brightness in a 1" x 1" patch at: 
        -new moon (phase = 0) 
        -full moon (phase = 14)

In astronomy, magnitudes are derived from flux densities using

$m = -2.5 \log_{10}\!\left(\frac{F_\nu}{F_{\nu,0}}\right),$

where $F_\nu$ is the source flux density and $F_{\nu,0}$ is the zero magnitude flux for that band. 

We're told that the star has the same $F_\nu$ in every filter: 

$F_\nu = 0.0364\ \text{mJy}$. 

We'll use the zero magnitude fluxes (in Jy) from Problem 1's table. 

First we must convert mJy to Jy: 
    
1 mJy = 10⁻³ Jy


So $F_\nu = 0.0364\,\text{mJy} = 3.64\times10^{-5}\,\text{Jy}$.

Next, we'll compute the magnitudes

$m = -2.5\log_{10}\!\left(\frac{3.64\times10^{-5}}{F_{\nu,0}}\right)$

for each band. 

Third, we must compare to the sky. Using the table from problem 2, we can find the magnitudes. We'll use the brightness ratio: 

$\frac{F_\text{star}}{F_\text{sky}} = 10^{0.4\,(m_\text{sky}-m_\text{star})}$.

We'll find this at the new and full moon for each filter. 

In [9]:
#known constants
F_star_mJy = 0.0364                  # given
F_star = F_star_mJy * 1e-3           # convert to Jy

filters = ["U", "B", "V", "R", "I"]
F_nu0 = [1810, 4260, 3640, 3080, 2550]  

# Sky brightness magnitudes
sky_new  = [22.0, 22.7, 21.8, 20.9, 19.9]
sky_full = [17.0, 19.5, 20.0, 19.9, 19.2]

In [10]:
# magnitudes of the star
m_star = [-2.5 * np.log10(F_star / F0) for F0 in F_nu0]

In [11]:
# brightness ratio vs sky
def flux_ratio(m_obj, m_sky):
    """Return F_obj / F_sky using magnitude difference."""
    return 10 ** (0.4 * (m_sky - m_obj))

ratio_new  = [flux_ratio(ms, mn) for ms, mn in zip(m_star, sky_new)]
ratio_full = [flux_ratio(ms, mf) for ms, mf in zip(m_star, sky_full)]

In [12]:
# assembling table
df3 = pd.DataFrame({
    "Filter": filters,
    "Fν0 (Jy)": F_nu0,
    "Star mag": np.round(m_star, 2),
    "Sky (New) mag": sky_new,
    "Sky (Full) mag": sky_full,
    "Star/New Ratio": np.round(ratio_new, 2),
    "Star/Full Ratio": np.round(ratio_full, 2)
})

df3

,Filter,Fν0 (Jy),Star mag,Sky (New) mag,Sky (Full) mag,Star/New Ratio,Star/Full Ratio
0,U,1810,19.24,22.0,17.0,12.69,0.13
1,B,4260,20.17,22.7,19.5,10.27,0.54
2,V,3640,20.00,21.8,20.0,5.25,1.00
3,R,3080,19.82,20.9,19.9,2.71,1.08
4,I,2550,19.61,19.9,19.2,1.30,0.68


### Problem 4

The transit of the exoplanet Kepler 90g was last observed on Julian Date 2460458.7241 ± 0.0033. This planet has an orbital period of 210.73514 ± 0.00017 days.
To calculate the calendar dates when the next three transits will occur if the planet’s orbit is stable and the period is constant, n-th future transit is simply: $$ t_n = t_0 + nP $$
n = 1, 2, 3

P = Period

Then, we can write the uncertainty as: 
$$
\sigma_{t_n} = \sqrt{ \sigma_{t_0}^2 + (n\sigma_P)^2 }
$$



- Reference mid-transit JD:2460458.7241 ± 0.0033
- Period: 210.73514 ± 0.00017 days

In [19]:
t0 = 2460458.7241      # Last observed transit in Julian Date
P = 210.73514          # Orbital period in days
sigma_t0 = 0.0033      # Uncertainty in JD
sigma_P = 0.00017      # Uncertainty in period (days)

n_list = [1, 2, 3]
results = []

for n in n_list:
    # Predict transit JD
    t_n = t0 + n * P
    sigma_tn = (sigma_t0**2 + (n * sigma_P)**2)**0.5 # Propagate timing uncertainty
    
    print(f"Transit {n}: JD = {t_n:.5f} ± {sigma_tn:.5f}")

Transit 1: JD = 2460669.45924 ± 0.00330
Transit 2: JD = 2460880.19438 ± 0.00332
Transit 3: JD = 2461090.92952 ± 0.00334


To convert JD into calender date, we used USNO Julian date converter - https://aa.usno.navy.mil/data/JulianDate.

2460669.45924 = A.D. 2024 December 24, 23:01:18.3 UT1 with 4.45min uncertainty

2460880.19438 = A.D. 2025 July 23,     16:39:54.4 UT1 with 4.47min uncertainty

2461090.92952 = A.D. 2026 February 19, 10:18:30.5 UT1 with 4.48min uncertainty


Using the KPNO observer calendar -https://noirlab.edu/science/observing-noirlab/observing-kitt-peak/ephemeris-kitt-peak-observers-calendar- and the celestial coordinates https://simbad.u-strasbg.fr/simbad/sim-id?Ident=Kepler+90&NbIdent=1&Radius=2&Radius.unit=arcmin&submit=submit+id
In KPNO local time (MST = UTC−7):

A.D. 2024 December 24, 23:01:18.3 UT1 = KPNO local: 2024-12-24 16:01:18 MST.

A.D. 2025 July 23,     16:39:54.4 UT1 = KPNO local: 2025-07-23 09:39:54 MST.

A.D. 2026 February 19, 10:18:30.5 UT1 = KPNO local: 2026-02-19 03:18:30 MST.


The key factor is given in the website above: Kepler-90 RA ≈ 18.96 h, Kitt Peak (KPNO) site: latitude ≈ 31.96° N, longitude ≈ −111.60° (West)

The Transit 1 (n = 1) occurs in 2024-12-24 16:01:18 MST, Altitude at transit (KPNO): ≈ 53.5° (high in the sky). LST at local midnight = 5.78 h
Although Kepler-90 is well above the horizon at the transit time, the event occurs in the local afternoon at Kitt Peak (16:01 MST), so it is not observable at night from KPNO on that date.

The Transit 2 (n = 2) occurs in 2025-07-23 09:39:54 MST, Altitude at transit (KPNO): ≈ −5.9° (below horizon). LST at local midnight = 19.645 h
Transit 2 is in the local morning (09:40 MST) and the target is below the horizon, so not observable at night

The Transit 3 (n = 3) occurs in 2026-02-19 03:18:30 MST, Altitude at transit (KPNO): ≈ 22.5° (above horizon). LST at local midnight ≈ 9.51 h
Transit 3 occurs during local night and Kepler-90 will be above the horizon at that time. It is therefore observable from KPNO, though the altitude is modest.

LST (local sidereal time) at local midnight tells which RA (right ascension) is on the meridian. If LST = RA of the star, that star is crossing the meridian, which is the best time to observe it because it is highest above the horizon. If RA < LST, the star has already crossed the meridian and is setting. If RA > LST, the star hasn’t crossed yet and is rising.

So, we can find out we have RA difference ≈ 9.45 h, which is consistent with it being east of the meridian in the early morning, but still above the horizon at about 03:18 MST. Therefore, we may plan a quick shortcut for which RAs are well placed that night using LST at local midnight.

**Observe this object using the 1.3m diameter telescope at MDM.
How fast will photons arrive from this object in the V filter, which has a width of 800 Angstroms?**

Photons hitting detector per second:
$$
N_{\mathrm{det}} = \Phi_{\star,\mathrm{band}} \times A \times \eta
$$
To get  $\Phi_{\star,\mathrm{band}}$, 
We have $F_\nu = F_\lambda \frac{\lambda^2}{c}$, rewrite as $f_\lambda = \frac{f_\nu \lambda^2}{c}$
,and $E_\gamma = \frac{hc}{\lambda}$ from problem 1 to convert V zero point to $F_\lambda$ and energy per photon at $\lambda$.

For the photon flux for a V=0 star:
$$
\Phi_0(\lambda) = \frac{f_{\lambda,0}}{E_\gamma}
$$

For the photon flux for Kepler-90 of magnitude 13.896:
$$
\Phi_{\star}(\lambda) = \Phi_{0} \times 10^{-0.4m_V}
$$


**Now we can calculate photons hitting detector per second with these parameters**:


Kepler-90 V magnitude: $m_V = 13.896$

central wavelength for V band: $\lambda = 5500\,\text{Å} = 5.5 \times 10^{-7}\,\mathrm{m}$

V zero point (spectral flux density):$f_{\nu,0} = 3631~\mathrm{Jy} = 3631 \times 10^{-26}\,\mathrm{W\,m^{-2}\,Hz^{-1}}$

Planck’s constant: $h = 6.626 \times 10^{-34}\,\mathrm{J\,s}$

Speed of light: $c = 2.998 \times 10^{8}\,\mathrm{m\,s^{-1}}$

Filter width: $\Delta\lambda = 800\,\text{Å}$

Telescope diameter: $D = 1.3\,\mathrm{m}$

Collecting area: $A = \pi \left( \frac{D}{2} \right)^2$

System efficiency: $\eta = 0.7$

In [20]:
import numpy as np

m_V = 13.896          # V magnitude of Kepler-90
lambda_c = 5500e-10   # central wavelength in m
delta_lambda = 800    # filter width in Å
eta = 0.7             # system efficiency
D = 1.3               # telescope diameter in m
h = 6.626e-34         # Planck constant in J·s
c = 2.998e8           # speed of light in m/s

# zero-point flux density for V band
fnu_0 = 3631e-26 # in W m^-2 Hz^-1
f_lambda_0 = fnu_0 * c / (lambda_c**2) * 1e-10  # in W m^-2 Å^-1

# Photon energy at central wavelength
E_gamma = h * c / lambda_c

# Photon flux for a V=0 star photons in s^-1 m^-2 Å^-1
phi_0 = f_lambda_0 / E_gamma

# Photon flux for Kepler-90
phi_star_perA = phi_0 * 10**(-0.4 * m_V)

# Integrate over filter width
phi_star = phi_star_perA * delta_lambda  # photons in s^-1 m^-2

# Telescope collecting area
A = np.pi * (D / 2)**2  # in m^2

# Detected photon rate
N_detected = phi_star * A * eta

# Print results
print(f"Photon arrival rate ≈ {N_detected:.2e} photons/s")
print(f" {N_detected * 60:.2e} photons/min")
print(f" {N_detected * 3600:.2e} photons/hour")

Photon arrival rate ≈ 2.05e+04 photons/s
 1.23e+06 photons/min
 7.37e+07 photons/hour


Using Kepler-90’s V-band magnitude and telescope parameters, we estimate a photon detection rate of approximately 2.05e+04 photons per second with the 1.3 m telescope and 70% system efficiency. This corresponds to about photons 1.23e+06 per minute, which provides a strong enough signal for precise photometric observations during transit events. This photon arrival rate indicates that even relatively faint stars like Kepler-90 ($m_V = 13.896$) can be effectively studied with medium-sized telescopes. Also, This calculation is essential for planning exposure times and ensuring the feasibility of observing campaigns at facilities like MDM.